pennylane/debugging.py

```diff
@@ -105,3 +108,75 @@ def get_snapshots(*args, **kwargs):
         return dbg.snapshots
 
     return get_snapshots
+
+
+class PLDB(pdb.Pdb):
+    """Custom debugging class integrated with Pdb."""
```

Comment:
To confirm, this isn't something users would typically interact with?

Reply:
Yes exactly! We don't want the user calling the PLDB class at all. In the same way we wouldn't want them to interact with the `QueuingManager` class for example.